feat: add sso (enterprise login) docs under platform (#10924)
supabase/supabase: apps/docs/pages/guides/platform/sso/sso-with-azure.mdx

import Layout from '~/layouts/DefaultGuideLayout'
import { Button, IconArrowDown } from 'ui'

export const meta = {
  id: 'sso-with-azure',

>   title: 'Setup SSO with Azure AD',
>   description: 'Configuration instructions for single sign-on with Azure AD (Entra)',

Reviewer: ```suggestion
  title: 'Set Up SSO with Azure AD',
  description: 'Configure single sign-on with Azure AD (Microsoft Entra).',
```

}

Supabase supports single sign-on (SSO) using Microsoft Azure AD.

## Step-by-step guide

> ### Step 1: Add and register an Enterprise Application

Reviewer: Adding custom anchors to make the URL a bit cleaner, easier to link to, and maintain in the future.
```suggestion
## Step 1: Add and register an Enterprise Application [#add-and-register-enterprise-application]
```

Open up the [Azure Active
Directory](https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/Overview)
dashboard for your Azure account.

Click the *Add* button then *Enterprise application*.

![Azure AD console: Default Directory Overview](/docs/img/sso-azure-step-01.png)

> ### Step 2: Choose Create your own application

Reviewer: ```suggestion
## Step 2: Choose Create your own application [#create-application]
```

You'll be using the custom enterprise application setup for Supabase.

![Azure AD console: Browse Azure AD Gallery, select: Create your own
application](/docs/img/sso-azure-step-02.png)

> ### Step 3: Fill in application details

Reviewer: ```suggestion
## Step 3: Fill in application details [#add-application-details] 
```

In the modal titled *Create your own application* enter the name you wish
Supabase to be available to your Azure AD users. `Supabase` works in most
cases.

Make sure to choose the third option: *Integrate any other application you
don't find in the gallery (Non-gallery)*.

![Azure AD console: Create your own application modal](/docs/img/sso-azure-step-03.png)

> ### Step 4: Choose the Set up single sign-on option

Reviewer: ```suggestion
## Step 4: Choose the Set up single sign-on option [#set-up-single-sign-on]
```

Before you get to assigning users and groups, which would allow accounts in
Azure AD to access Supabase, you need to configure the SAML details that allows
Supabase to accept sign in requests from Azure AD.

![Azure AD console: Supabase custom enterprise application, selected Set up
single sign-on](/docs/img/sso-azure-step-04.png)

> ### Step 5: Select SAML single sign-on method

Reviewer: ```suggestion
## Step 5: Select SAML single sign-on method [#saml-sso]
```

Supabase only supports the SAML 2.0 protocol for Single Sign-On, which is an
industry standard.

![Azure AD console: Supabase application, Single sign-on configuration screen,
selected SAML](/docs/img/sso-azure-step-05.png)

> ### Step 6: Upload SAML-based Sign-on metadata file

Reviewer: ```suggestion
## Step 6: Upload SAML-based Sign-on metadata file [#upload-saml-metadata]
```

First you need to download Supabase's SAML metadata file. Click the button
below to initiate a download of the file.

<a href="https://alt.supabase.io/auth/v1/sso/saml/metadata?download=true"><Button size="large" icon={<IconArrowDown />}>Download Supabase SAML Metadata File</Button></a>

Alternatively, visit this page to initiate a download: `https://alt.supabase.io/auth/v1/sso/saml/metadata?download=true`

Click on the *Upload metadata file* option in the toolbar and select the file
you just downloaded.

![Azure AD console: Supabase application, SAML-based Sign-on screen,
selected Upload metadata file button](/docs/img/sso-azure-step-06-1.png)

All of the correct information should automatically populate the *Basic SAML
Configuration* screen as shown.

![Azure AD console: Supabase application, SAML-based Sign-on screen,
Basic SAML Configuration shown](/docs/img/sso-azure-step-06-2.png)

**Make sure you input these additional settings.**

| Setting | Value |
| ------- | ----- |
| Sign on URL | `https://app.supabase.com/sign-in-sso` |
| Relay State | `https://app.supabase.com` |

Finally, click the *Save* button to save the configuration.

> ### Step 7: Obtain metadata URL and send to Supabase

Reviewer: ```suggestion
## Step 7: Obtain metadata URL and send to Supabase [#send-metadata-url]
```

Supabase needs to finalize enabling single sign-on with your Azure AD
application. To do this, please copy and send the link under **App Federation
Metadata Url** in *section 3 **SAML Certificates*** to your support
contact and await further instructions. If you're not clear who to send this
link to or need further assistance, please reach out to
[support@supabase.com](mailto:support@supabase.com).

**Do not test the login until you have heard back from the support contact.**

![Azure AD console: Supabase application, SAML Certificates card
shown, App Federation Metadata Url highlighted](/docs/img/sso-azure-step-07.png)

> ### Step 8: Wait for confirmation

Reviewer: ```suggestion
## Step 8: Wait for confirmation [#confirmation]
```

Please wait for confirmation or further instructions from your support contact
at Supabase before proceeding to the next step. It usually takes us 1 business
day to configure SSO for you.

> ### Step 9: Test single sign-on

Reviewer: ```suggestion
## Step 9: Test single sign-on [#testing]
```

*Testing sign-on before your Azure AD has been registered with Supabase will
not work. Make sure you've received confirmation from your support contact at

> Supabase as laid out in [Step 8](#step-8-wait-for-confirmation).*

Reviewer: ```suggestion
Supabase as laid out in the [confirmation](#confirmation) step.*
```
